bears/rust/RustClippyLintBear.py

```diff
@@ -0,0 +1,71 @@
+import json
+import os
+
+from coalib.bearlib.abstractions.Linter import linter
+from coalib.bears.GlobalBear import GlobalBear
+from coalib.misc.Shell import run_shell_command
+from coalib.results.Result import Result
+from coalib.results.RESULT_SEVERITY import RESULT_SEVERITY
+from dependency_management.requirements.CargoRequirement import (
+        CargoRequirement)
+
+
+@linter(executable='cargo',
+        output_format=None,
+        global_bear=True,
+        use_stderr=True)
+class RustClippyLintBear(GlobalBear):
+    LANGUAGES = {'Rust'}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    CAN_DETECT = {
+        'Formatting',
+        'Unused Code',
+        'Syntax',
+        'Unreachable Code',
+        'Smell',
+        'Code Simplification',
+    }
+    EXECUTABLE = 'cargo.exe' if os.name == 'nt' else 'cargo'
+    REQUIREMENTS = {
+        CargoRequirement('clippy')
+    }
+    SEVERITY_MAP = {
+        'warning': RESULT_SEVERITY.NORMAL,
+        'error': RESULT_SEVERITY.MAJOR,
+    }
+
+    @staticmethod
+    def create_arguments(config_file):
+        args = ('clippy', '--quiet', '--color', 'never',
+                '--', '-Z', 'unstable-options',
+                '--error-format', 'json',
+                '--test')
+        return args
+
+    def process_output(self, output, filename, file):
+        _, stderr_output = output
```

Comment:
when you deactivate stdout with `use_stdout=False`, you don't need to unpack here, `output` will contain `stderr` itself